components-core/src/main/java/com/adyen/checkout/components/core/internal/ui/model/ComponentParamsMapper.kt

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2024 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by josephj on 7/2/2024.
+ */
+
+package com.adyen.checkout.components.core.internal.ui.model
+
+import androidx.annotation.RestrictTo
+import com.adyen.checkout.components.core.Amount
+import com.adyen.checkout.components.core.CheckoutConfiguration
+import com.adyen.checkout.core.Environment
+import java.util.Locale
+
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+interface ComponentParamsMapper {
+    val checkoutConfiguration: CheckoutConfiguration
+    val deviceLocale: Locale
+    val dropInOverrideParams: DropInOverrideParams?
+    val componentSessionParams: SessionParams?
+}
+
+val ComponentParamsMapper.sessionParams: SessionParams?
+    get() = dropInOverrideParams?.sessionParams ?: componentSessionParams
+
+val ComponentParamsMapper.shopperLocale: Locale
+    get() = checkoutConfiguration.shopperLocale ?: deviceLocale
+
+// drop-in amount has priority over sessions because of partial payments
+// after a partial payment is successful the amount should be updated to the remaining amount
+// but the backend is not returning the correct value with the sessions setup call
+// therefore we modify the amount ourselves in drop-in
+val ComponentParamsMapper.amount: Amount?
```

Comment:
Behaviorally nothing actually changed, we were manually overriding both the sessions amount and drop-in amount ([check this line](https://github.com/Adyen/adyen-android/blob/796b359d30e0b2ccc2792d673aa19200ff2d36b3/drop-in/src/main/java/com/adyen/checkout/dropin/internal/ui/DropInViewModel.kt#L318).

With this change I am now using the [overrideAmount](https://github.com/Adyen/adyen-android/blob/c89d024606d252f5a52726e81fb877f3b4c9016e/drop-in/src/main/java/com/adyen/checkout/dropin/internal/ui/DropInViewModel.kt#L79`) variable to achieve that but I update the amount in `DropInParams` instead of `SessionDetails`. Do you think that still needs improvement? I can maybe still manually override the amount in `SessionDetails` with the amount from `DropInParams` but I am not so sure about that. WDYT?

Reply:
Hmm let's discuss it offline

Reply:
Fixed as discussed